homeassistant/components/wled/light.py

```diff
@@ -0,0 +1,222 @@
+"""Support for LED lights."""
+import logging
+from typing import Any, Callable, List, Optional, Tuple
+
+from wled import WLED, Effect, WLEDError
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_EFFECT,
+    ATTR_HS_COLOR,
+    ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_COLOR_TEMP,
+    SUPPORT_EFFECT,
+    SUPPORT_TRANSITION,
+    Light,
+)
+from homeassistant.components.wled import WLEDDeviceEntity
+from homeassistant.components.wled.const import (
+    ATTR_COLOR_PRIMARY,
+    ATTR_INTENSITY,
+    ATTR_ON,
+    ATTR_PALETTE,
+    ATTR_PLAYLIST,
+    ATTR_PRESET,
+    ATTR_SEGMENT_ID,
+    ATTR_SPEED,
+    DATA_WLED_CLIENT,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+import homeassistant.util.color as color_util
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up WLED light based on a config entry."""
+    wled: WLED = hass.data[DOMAIN][entry.entry_id][DATA_WLED_CLIENT]
+
+    # Does the WLED device support RGBW
+    rgbw = wled.device.info.leds.rgbw
+
+    # List of supported effects
+    effects = wled.device.effects
+
+    # WLED supports splitting a strip in multiple segments
+    # Each segment will be a separate light in Home Assistant
+    lights = []
+    for light in wled.device.state.segments:
+        lights.append(WLEDLight(entry.entry_id, wled, light.segment_id, rgbw, effects))
+
+    async_add_entities(lights, True)
+
+
+class WLEDLight(Light, WLEDDeviceEntity):
+    """Defines a WLED light."""
+
+    def __init__(
+        self, entry_id: str, wled: WLED, segment: int, rgbw: bool, effects: List[Effect]
+    ):
+        """Initialize WLED light."""
+        self._effects = effects
+        self._rgbw = rgbw
+        self._segment = segment
+
+        self._brightness: Optional[int] = None
+        self._color: Optional[Tuple[float, float]] = None
+        self._effect: Optional[str] = None
+        self._state: Optional[bool] = None
+
+        # Only apply the segment ID if it is not the first segment
+        name = wled.device.info.name
+        if segment != 0:
+            name += f" {segment}"
+
+        super().__init__(entry_id, wled, name, "mdi:led-strip-variant")
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return (
+            f"{DOMAIN}_{self.wled.device.info.mac_address}_light_{str(self._segment)}"
```

Comment:
The entity registry is aware of both the integration domain and the platform domain. We can remove the domain prefix and "light".

Reply:
We shouldn't have to copy the segment to string I think.